### PR TITLE
ci: add workflow to open PR after release

### PR DESCRIPTION
**Pull request by obycode**
Automatically merge main back into develop after a release.

### CODE REVIEW

File: .github/workflows/open-develop-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    types:
+      - released
+  workflow_dispatch:
+
+jobs:
+  run:
+    name: Upsteam develop branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Clarinet
```
Comment (CharlieC3) [inline .github/workflows/open-develop-pr.yaml:18]:
```suggestion
      - name: Checkout clarity-repl
```

File: .github/workflows/open-develop-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          submodules: recursive
+
+      - name: Open pull request
+        uses: repo-sync/pull-request@v2
+        with:
+          source_branch: main
+          destination_branch: develop
+          pr_title: "chore: update develop branch"
+          pr_body: |
+            :robot: This is an automated pull request created from a new release in [clarinet](https://github.com/hirosystems/clarinet/releases).
```
Comment (CharlieC3) [inline .github/workflows/open-develop-pr.yaml:30]:
```suggestion
            :robot: This is an automated pull request created from a new release in [clarity-repl](https://github.com/hirosystems/clarity-repl/releases).
```

**Review by CharlieC3 — CHANGES_REQUESTED** [review]
Couple minor changes

File: .github/workflows/open-develop-pr.yaml (hunk 1 above)
Comment (obycode) [reply .github/workflows/open-develop-pr.yaml]: Fixed in 1fe39ff438e7da97171d6a7e4685854debaddea5.

File: .github/workflows/open-develop-pr.yaml (hunk 2 above)
Comment (obycode) [reply .github/workflows/open-develop-pr.yaml]: Fixed in 1fe39ff438e7da97171d6a7e4685854debaddea5. Thanks Charlie!

**Review by CharlieC3 — APPROVED** [review]
Good to go! :shipit:

**Comment by obycode** [comment]
I just pushed an update to rebase on the latest. If the CI tests all pass, I will merge it in. Thanks @CharlieC3.

**Pull request merged**